common/objectcachev2.h

```diff
@@ -120,37 +114,32 @@ class ObjectCacheV2 {
         SCOPED_LOCK(maplock);
         auto it = map.find(keyitem);
         if (it == map.end()) {
-            // item = new Box(std::forward<KeyType>(key));
-            auto rt = map.emplace(key);
+            auto rt = map.emplace(std::forward<KeyType>(key));
             if (rt.second) item = (Box*)&*rt.first;
-        } else
+        } else {
             item = (Box*)&*it;
+        }
         assert(item);
         item->timestamp = photon::now;
-        cycle_list.pop(item);
-        cycle_list.push_back(item);
+        lru_list.pop(item);
+        lru_list.push_back(item);
         return *item;
     }
 
     uint64_t __expire() {
-        intrusive_list<Box> delete_list;
         uint64_t now = photon::now;
-        {
-            SCOPED_LOCK(maplock);
-            if (cycle_list.empty()) return 0;
-            if (photon::sat_add(cycle_list.front()->timestamp, lifespan) >
-                now) {
-                return photon::sat_add(cycle_list.front()->timestamp,
-                                       lifespan) -
-                       now;
-            }
-            auto x = cycle_list.front();
-            while (x && (photon::sat_add(x->timestamp, lifespan) < now)) {
-                cycle_list.pop(x);
-                __update(*x, nullptr);
-                map.erase(*x);
-                x = cycle_list.front();
-            }
+        SCOPED_LOCK(maplock);
+        if (lru_list.empty()) return 0;
+        if (photon::sat_add(lru_list.front()->timestamp, lifespan) > now) {
+            return photon::sat_add(lru_list.front()->timestamp, lifespan) -
+                    now;
+        }
+        auto x = lru_list.front();
+        while (x && (photon::sat_add(x->timestamp, lifespan) < now)) {
+            lru_list.pop(x);
+            __update(*x, nullptr);
```

Comment:
it's more performant to realize x->reset(), which simply invokes shared_ptr::reset().

Reply:
In new implementation still use exchange/swap like stuff instead of reset().

swapped out object still have at lease one reference if result is assigned to another variable, so it could able to control when to perform object destruction.